lib/utils/remote-build.ts

```diff
@@ -210,6 +213,28 @@ async function cancelBuildIfNecessary(build: RemoteBuild): Promise<void> {
 	}
 }
 
+/**
```

Comment:
I'm thinking that this doesn't necessarily only apply to remote-builds, right? Especially if it's added to `resin-multibuild`, we can get this on local mode devices for free. I definitely think that we should move this to a less specific location and use it in both remote-build and local-build.

Reply:
You mean for example the following command, right?
```
balena push 192.168.2.3 --registry-secrets sample_secrets.json -s secrets_app
```
I was looking at the implementation of push-to-IP-address and, if I got it right, the CLI opens a connection to the docker daemon (balena engine) of a local-mode device and uploads the tar stream directly to the daemon. In this case, I think we don't want a `.balena` metadata folder to be inserted in the stream. Instead, it might be enough for the CLI itself to add the `registryconfig` option to Dockerode -- which is what the cloud builder does _after_ extracting the `.balena` metadata folder from the tar stream. If this is right, then actually this `getTarStreamCallbackForRegistrySecrets()` function applies only to remote builds.

Reply:
I've now put together a private registry with a self-signed SSL certificate for testing and I can confirm that pushing to a device IP address works with the implementation as it is!
(The cloud builder is OK with pulling from a private registry over plain HTTP, but the balena-engine daemon on a device insists on HTTPS.)